react/week4/src/components/GetUsersData.js

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useContext } from "react";
+import { userListContext } from "../App";
+import SearchLoader from "./SearchLoader";
+import UserList from "./UserList";
+
+function GetUsersData() {
+  const { setUser, input, error, setError, loading, setLoading } = useContext(
+    userListContext
+  );
+  useEffect(() => {
+    const SEARCH_URL = `https://api.github.com/search/users?q=${input}`;
+    (async () => {
+      try {

```

Comment:
We need to guard the fetching with the check below. This way we don't call the API with an empty string.
```js
if (input !== "") {
    // do the request
    // bla bla
}
```

Reply:
corrected